explorer/src/providers/stats/solanaDashboardInfo.tsx

```diff
@@ -0,0 +1,96 @@
+import { PerfSample } from "@solana/web3.js";
+import { ClusterStatsStatus } from "./solanaClusterStats";
+
+export type DashboardInfo = {
+  status: ClusterStatsStatus;
+  avgBlockTime_1h: number;
+  avgBlockTime_1min: number;
+  epochInfo: {
+    absoluteSlot: number;
+    blockHeight: number;
+    epoch: number;
+    slotIndex: number;
+    slotsInEpoch: number;
+  };
+};
+
+export enum DashboardInfoActionType {
+  SetPerfSamples,
+  SetEpochInfo,
+  SetError,
+  Reset,
+}
+
+export type DashboardInfoActionSetPerfSamples = {
+  type: DashboardInfoActionType.SetPerfSamples;
+  data: PerfSample[];
+};
+
+export type DashboardInfoActionSetEpochInfo = {
+  type: DashboardInfoActionType.SetEpochInfo;
+  data: any;
+};
+
+export type DashboardInfoActionReset = {
+  type: DashboardInfoActionType.Reset;
+  data: DashboardInfo;
+};
+
+export type DashboardInfoActionSetError = {
+  type: DashboardInfoActionType.SetError;
+  data: string;
+};
+
+export type DashboardInfoAction =
+  | DashboardInfoActionSetPerfSamples
+  | DashboardInfoActionSetEpochInfo
+  | DashboardInfoActionReset
+  | DashboardInfoActionSetError;
+
+export function dashboardInfoReducer(
+  state: DashboardInfo,
+  action: DashboardInfoAction
+) {
+  const status =
+    state.avgBlockTime_1h !== 0 && state.epochInfo.absoluteSlot !== 0
+      ? ClusterStatsStatus.Ready
+      : ClusterStatsStatus.Loading;
+
+  switch (action.type) {
+    case DashboardInfoActionType.SetPerfSamples:
+      const samples = action.data.map((sample) => {
+        return sample.samplePeriodSecs / sample.numSlots;
+      });
+
+      const avgBlockTime_1h =
+        samples.reduce((sum: number, cur: number) => {
+          return sum + cur;
+        }, 0) / 60;
+
+      return {
+        ...state,
+        avgBlockTime_1h,
+        avgBlockTime_1min: samples[0],
+        status,
+      };
+    case DashboardInfoActionType.SetEpochInfo:
+      return {
+        ...state,
+        epochInfo: action.data,
+        status,
+      };
+    case DashboardInfoActionType.SetError:
+      return {
+        ...state,
+        status: ClusterStatsStatus.Error,
+      };
+    case DashboardInfoActionType.Reset:
+      return {
+        ...action.data,
+      };
+    default:
+      return {
+        ...state,
+      };
```

Comment:
If nothing changed, you should return `state` directly in order to avoid causing unnecessary renders